packages/client/src/views/Home/RecordAudit.tsx

```diff
@@ -0,0 +1,608 @@
+  type: 'No event',
+  trackingId: 'No tracking id',
+  dateOfBirth: 'No date of birth',
+  dateOfDeath: 'No date of death',
+  placeOfBirth: 'No place of birth',
+  placeOfDeath: 'No place of death',
+  informant: 'No informant'
+}
+
+const getCaptitalizedword = (word: string | undefined): string => {
+  word = word && word.toUpperCase()[0] + word.toLowerCase().slice(1)
+  return word || ''
+}
+
+const isBirthApplication = (
+  application: GQLEventSearchSet | null
+): application is GQLBirthEventSearchSet => {
+  return (application && application.type === 'Birth') || false
+}
+
+const isDeathApplication = (
+  application: GQLEventSearchSet | null
+): application is GQLDeathEventSearchSet => {
+  return (application && application.type == 'Death') || false
+}
+
+const getDraftApplicationName = (application: IApplication): string => {
+  let name = ''
+  const applicationName = application.data.child || application.data.deceased
+  if (applicationName) {
+    if (applicationName.firstNamesEng && applicationName.familyNameEng) {
+      name = applicationName.firstNamesEng + ' ' + applicationName.familyNameEng
+    } else {
+      name =
+        (applicationName.firstNamesEng as string) ||
+        (applicationName.familyNameEng as string) ||
+        ''
+    }
+  }
+  return name
+}
+
+const getWQApplicationName = (application: GQLEventSearchSet): string => {
+  let name = ''
+  const applicationName =
+    (isBirthApplication(application) &&
+      application.childName &&
+      application.childName[0]) ||
+    (isDeathApplication(application) &&
+      application.deceasedName &&
+      application.deceasedName[0])
+
+  if (applicationName) {
+    if (applicationName.firstNames && applicationName.familyName) {
+      name = applicationName.firstNames + ' ' + applicationName.familyName
+    } else {
+      name = applicationName.familyName || applicationName.firstNames || ''
+    }
+  }
+  return name
+}
+
+const getGQLApplicationName = (application: IGQLApplication): string => {
+  let name = ''
+  const applicationName =
+    application && application.child && application.child.name
+      ? application.child.name[0]
+      : application.deceased &&
+        application.deceased.name &&
+        application.deceased.name[0]
+  if (applicationName) {
+    if (
+      (applicationName as GQLHumanName).firstNames &&
+      (applicationName as GQLHumanName).familyName
+    ) {
+      name =
+        (applicationName as GQLHumanName).firstNames +
+        ' ' +
+        (applicationName as GQLHumanName).familyName
+    } else {
+      name =
+        (applicationName as GQLHumanName).familyName ||
+        (applicationName as GQLHumanName).firstNames ||
+        ''
+    }
+  }
+  return name
+}
+
+const getLocation = (application: IApplication, props: IFullProps) => {
+  let locationType = ''
+  let locationId = ''
+  let locationDistrict = ''
+  let locationPermanent = ''
+  if (
+    application.event === 'death' &&
+    application.data &&
```

Comment:
I'm not sure if our current TS version already supports this, but could you try replacing these 

`application.data && application.data.deathEvent`

with 

`application.data?.deathEvent`

or 

```ts
locationType =
      (application.data.deathEvent.deathPlaceAddress &&
        application.data.deathEvent.deathPlaceAddress.toString()) ||
      ''
```

with 

```ts
locationType = application.data.deathEvent.deathPlaceAddress?.toString() || ''
```